### ISSUE TITLE
how to get hyperscan match result immediatly, without callback function?

### ISSUE THREAD

**Issue by wy7980** [intel/hyperscan#180]
Is there anaway to get if the target matched or not immediatly?

**Comment by xiangwang1** [comment]
You can add single match flag `HS_FLAG_SINGLEMATCH` at compile time. The scan will terminate once a match is found.

**Comment by wy7980** [comment]
> You can add single match flag `HS_FLAG_SINGLEMATCH` at compile time. The scan will terminate once a match is found.

Yes, but it seems thath I must use the callback function to get it matched or not?

**Comment by xiangwang1** [comment]
Yes, you have to use callback function. Any specific reason why you want to avoid callback?

**Comment by wy7980** [comment]
> Yes, you have to use callback function. Any specific reason why you want to avoid callback?

No, just feel more convenient to get result from return value of a function instead of one more callback function.

**Comment by ryankang95** [comment]
Hi, @xiangwang1 . i want to get the start offset of match and the end offset of match, but these params only in callback function, what can i do to get these offset?
For example, i have one string "hello 2018, by 2017",  i want to get two substring {'2017', '2018'} and get their's index {[6,9], [14,17]}. what should i do ?  thanks!

**Comment by xiangwang1** [comment]
Already answered at https://github.com/intel/hyperscan/issues/189.
